shell/browser/api/electron_api_session.cc

```diff
@@ -912,7 +912,7 @@ static void StartPreconnectOnUI(ElectronBrowserContext* browser_context,
                                 int num_sockets_to_preconnect) {
   std::vector<predictors::PreconnectRequest> requests = {
       {url::Origin::Create(url), num_sockets_to_preconnect,
-       net::NetworkIsolationKey()}};
+       net::NetworkIsolationKey::CreateTransient()}};
```

Comment:
Hm, is this correct? I don't know much about network isolation but it seems like preconnect might respond poorly to having a different network isolation key between the preconnect and the connect?

Reply:
@deepak1556 can you respond to @nornagon's question here?  If there is an issue with this, I can followup in #30676.

Reply:
Yeah you are right, I should have created the isolation key from the origin of preconnect url. The current implementation will definitely disallow sharing the sockets created in the preconnect step.